fabtests/unit/mr_cache_evict.c

```diff
@@ -317,12 +400,15 @@ static void *mem_alloc(enum alloc_type type)
 	case BRK:
 		ptr = brk_alloc();
 		break;
+	case CUDA:
+		ptr = cuda_malloc();
+		break;
 	default:
 		return NULL;
 	}
 
 	if (ptr) {
-		if (geteuid() == 0) {
+		if (geteuid() == 0 && type != CUDA) {
```

Comment:
The order of the checks can be reversed.  No need to call geteuid() on non-system memory.  Also, please change the type != CUDA to a check against iface == FI_HMEM_SYSTEM.  See below for request to create a function for that. E.g.
```
if (get_iface(type) == FI_HMEM_SYSTEM && geteuid() == 0) {
```